Content.Shared/Interaction/SharedInteractionSystem.cs

```diff
@@ -253,6 +254,19 @@ private bool ShouldCheckAccess(EntityUid user)
 
             if (!altInteract && TryComp(user, out SharedCombatModeComponent? combatMode) && combatMode.IsInCombatMode)
             {
+                if(target == null) return;
+
+                if(!TryComp<ItemComponent>(target.Value, out _)) return;
+
+                // We need this because of funny SharedGunSystem realization
+                if (TryComp<GunComponent>(target.Value, out var gunComponent))
+                {
+                    gunComponent.NextFire = _gameTiming.CurTime + TimeSpan.FromMilliseconds(125);
+                    Dirty(gunComponent);
+                }
+
+                InteractHand(user, target.Value);
```

Comment:
Before I merge it, why exactly was this needed?

Reply:
If this is not added, the weapon will fire as soon as you pick it up